app/controllers/steps_controller.rb

```diff
@@ -97,11 +97,6 @@ def create
           status: :ok
         end
       else
-        # On error, delete the newly added files from S3, as they were
-        # uploaded on client-side (in case of client-side hacking of
-        # asset's signature response)
-        Asset.destroy_all(new_assets)
```

Comment:
Probably, but anyway there is no 'new_assets' in the code, so it will not work in any case.